keras/activations.py

```diff
@@ -7,13 +7,9 @@ def softmax(x):
     if ndim == 2:
         return K.softmax(x)
     elif ndim == 3:
-        # apply softmax to each timestep
-        def step(x, states):
-            return K.softmax(x), []
-        last_output, outputs, states = K.rnn(step, x,
-                                             [],
-                                             mask=None)
-        return outputs
+        e = K.exp(x)
```

Comment:
I believe this is prone to overflowing. It should be something like `e = K.exp(x - K.max(x, axis=-1, keepdims=True))`.